source/framework/vx_array.c

```diff
@@ -22,6 +22,182 @@ static void ownInitArrayObject(
     vx_array arr, vx_enum item_type, vx_size capacity, vx_bool is_virtual);
 static vx_size ownGetArrayItemSize(vx_context context, vx_enum item_type);
 static vx_bool ownIsValidArrayItemType(vx_context context, vx_enum item_type);
+static vx_bool ownIsValidInputAndOutputArrays(vx_array input, vx_array output);
```

Comment:
In general, this is a lot of new code added for every data object which is doing mostly the same thing.  As part of avoiding growing the footprint, and also helping with code coverage, I think it is worth looking for ways to generalize much of these functions to using "references" instead of specific objects.  We have already done something similar with the allocation and release callbacks (see: ownAllocReferenceBufferGeneric, and ownReleaseReferenceBufferGeneric ... these are not used for all data objects, but can be used for most of them that follow the same pattern).  I will add a few specific comments in specific places, using the GENERIC keyword to allude back to this comment.

Reply:
we see the generic function mostly for the mem allocator and destructor callback. 
If I correctly understand here, you want us to make a proposal for the ownIsvalid functions?
if we do a generic function, we need to roll it out then, who is doing that?